app/validation/validator.py

```diff
@@ -729,9 +729,49 @@ def _validate_routing_rule(self, rule, answer_ids_with_group_id, block_or_group)
                     rule["when"], answer_ids_with_group_id, block_or_group["id"]
                 )
             )
+            for when_dict in rule.get("when"):
+                errors.extend(
+                    self._validate_routing_when_dict(
+                        when_dict, answer_ids_with_group_id
+                    )
+                )
 
         return errors
 
+    def _validate_routing_when_dict(self, when_dict, answer_ids_with_parent_id):
+        errors = []
+        when_values = when_dict.get("values", [])
+        when_value = when_dict.get("value")
+
+        if when_value:
+            when_values.append(when_value)
+
+        for rule_value in when_values:
+            valid_rule = self.is_rule_value_valid(answer_ids_with_parent_id, rule_value)
+
+            if not valid_rule:
+                errors.append(
+                    Validator._error_message(
+                        f"Answer option and routing rule values mismatch, "
+                        f"missing answer value: {rule_value}"
+                    )
+                )
+
+        return errors
+
+    @staticmethod
+    def is_rule_value_valid(answer_ids_with_parent_id, rule_value):
+        if isinstance(rule_value, int):
+            return True
```

Comment:
I think this check would be better placed in the parent method.

Reply:
Placed it in the parent now. This changed the numbers test to assert "false" now.